docs/gallery/advanced_io/streaming.py

```diff
@@ -1,98 +1,119 @@
-# If you have not already, install the latest release of the ``dandi`` package.
-#
-# .. code-block:: bash
-#
-#   pip install dandi
-#
-# .. code-block:: python
-#
-#   from dandi.dandiapi import DandiAPIClient
-#
-#   dandiset_id = '000006'  # ephys dataset from the Svoboda Lab
-#   filepath = 'sub-anm372795/sub-anm372795_ses-20170718.nwb'  # 450 kB file
-#   with DandiAPIClient() as client:
-#       asset = client.get_dandiset(dandiset_id, 'draft').get_asset_by_path(filepath)
-#       s3_path = asset.get_content_url(follow_redirects=1, strip_query=True)
-
-####################
-# Finally, instantiate a :py:class:`~pynwb.NWBHDF5IO` object with the S3 URL and specify the driver as "ros3". This
-# will download metadata about the file from the S3 bucket to memory. The values of datasets are accessed lazily,
-# just like when reading an NWB file stored locally. So, slicing into a dataset will require additional time to
-# download the sliced data (and only the sliced data) to memory.
-#
-# .. code-block:: python
-#
-#   from pynwb import NWBHDF5IO
-#
-#   with NWBHDF5IO(s3_path, mode='r', load_namespaces=True, driver='ros3') as io:
-#       nwbfile = io.read()
-#       print(nwbfile)
-#       print(nwbfile.acquisition['lick_times'].time_series['lick_left_times'].data[:])
-
-####################
-# Method 2: s3fs
-# ~~~~~~~~~~~~~~
-# s3fs is a library that creates a virtual filesystem for an S3 store. With this approach, a virtual file is created
-# for the file and virtual filesystem layer will take care of requesting data from the s3 bucket whenever data is
-# read from the virtual file.
-#
-# First install s3fs:
-#
-# .. code-block:: bash
-#
-#   pip install s3fs
-#
-# Then in Python:
-#
-# .. code-block:: python
-#
-#   import s3fs
-#   import pynwb
-#   import h5py
-#
-#   fs = s3fs.S3FileSystem(anon=True)
-#
-#   f = fs.open("s3://dandiarchive/blobs/43b/f3a/43bf3a81-4a0b-433f-b471-1f10303f9d35", 'rb')
-#   file = h5py.File(f)
-#   io = pynwb.NWBHDF5IO(file=file, load_namespaces=True)
-#
-#   io.read()
-#
-# The above snippet opens an arbitrary file on DANDI. You can use the ``DandiAPIClient`` to find the s3 path,
-# but you will need to adjust this url to give it a prefix of "s3://dandiarchive/" as shown above.
-#
-# The s3fs approach has the advantage of being more robust that ROS3. Sometimes s3 requests are interrupted,
-# and s3fs has internal mechanisms to retry these requests automatically, whereas ROS3 does not. However, it may not
-# be available on all platforms. s3fs does not currently work for Windows.
+   with NWBHDF5IO(s3_url, mode='r', load_namespaces=True, driver='ros3') as io:
+       nwbfile = io.read()
+       print(nwbfile)
+       print(nwbfile.acquisition['lick_times'].time_series['lick_left_times'].data[:])
 
+Streaming Method 2: fsspec
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+fsspec is another data streaming approach that quite flexible and has several performance advantages. This library
```

Comment:
```suggestion
fsspec is another data streaming approach that is quite flexible and has several performance advantages. This library
```

Reply:
Can you also add a link to the ffspec documentation here.

Reply:
I found the phase ``and has several performance advantages.`` a bit confusing, mainly because it is non-specific. Instead, I think it would be useful if we could include a more specific section on the advantages/disadvantages of the two options and which option to use when.